test/test_testing.py

```diff
@@ -4073,6 +4073,295 @@ def test_add_layer_merge(self, client: _TestingPebbleClient):
+                    tcp:
+                        port: 8082
+            """,
+            combine=True,
+        )
+        # key2 should be added to environment in up check
+        # header2 should be added to headers in ready-http check
+        # port should be changed to 8082 in ready-tcp check
+        # all other properties should remain the same
+        plan = client.get_plan()
+        assert (
+            textwrap.dedent("""\
+            checks:
+              ready-http:
+                http:
+                  headers:
+                    header1: value1
+                    header2: value2
+                  url: https://example.com:3100/health
+                level: ready
+                override: merge
+                period: 30s
+                threshold: 1
+              ready-tcp:
+                level: ready
+                override: merge
+                period: 30s
+                tcp:
+                  host: localhost
+                  port: 8082
+                threshold: 1
+              up:
+                exec:
+                  command: service nginx status 1
+                  environment:
+                    key1: value1
+                    key2: value2
+                level: alive
+                override: merge
+                period: 30s
+                threshold: 1
+            """)
+            == plan.to_yaml()
+        )
+
+    def test_add_layer_log_targets_combine_override_replace(self, client: _TestingPebbleClient):
+        plan = client.get_plan()
+        assert isinstance(plan, pebble.Plan)
+        assert plan.to_yaml() == '{}\n'
+        client.add_layer(
+            'foo',
+            pebble.Layer("""\
+            log-targets:
+                baz:
+                    override: replace
+                    type: loki
+                    location: https://example.com:3100/loki/api/v1/push
+                    services:
+                        - foo
+                    labels:
+                        key: value
+                        key1: value1
+            """),
+        )
+        client.add_layer(
+            'foo',
+            pebble.Layer("""\
+            log-targets:
+                baz:
+                    override: replace
+                    type: loki
+                    location: https://example123.com:3100/loki/api/v1/push
+                    services:
+                        - foo
+                    labels:
+                        key1: value1
+            """),
+            combine=True,
+        )
+        plan = client.get_plan()
+        assert (
+            textwrap.dedent("""\
+            log-targets:
+              baz:
+                labels:
+                  key1: value1
+                location: https://example123.com:3100/loki/api/v1/push
+                override: replace
+                services:
+                - foo
+                type: loki
+            """)
+            == plan.to_yaml()
+        )
```

Comment:
This is asserting that the plan is equal to the second layer, right? Because `override: replace` was used.

I think the test would be clearer (and more resilient) if it made that explicit. Something like (untested):

```suggestion
        layer = pebble.Layer("""\
            log-targets:
                baz:
                    override: replace
                    type: loki
                    location: https://example123.com:3100/loki/api/v1/push
                    services:
                        - foo
                    labels:
                        key1: value1
            """
        )
        client.add_layer(
            'foo',
            layer,
            combine=True,
        )
        plan = client.get_plan()
        assert layer.to_yaml() == plan.to_yaml()
```

Reply:
I dont have a strong opinion about it so I don't mind changing it but to me the existing code looks explicit too :)
I follow the pattern in the test_add_layer_combine_override_replace. I'm afraid that changing just one of them would look confusing.